### PR TITLE
Add optional timeout for app injection process

### PR DESCRIPTION
**Pull request by disaykin**

### CODE REVIEW

**Comment by disaykin** [comment]
Adding APT Sources
121.36s$ sudo -E apt-add-repository -y "ppa:ubuntu-toolchain-r/test"
gpg: keyring `/tmp/tmpje0_cwne/secring.gpg' created
gpg: keyring `/tmp/tmpje0_cwne/pubring.gpg' created
gpg: requesting key BA9EF27F from hkp server keyserver.ubuntu.com
Error: retrieving gpg key timed out.
gpg: keyserver timed out
gpg: keyserver receive failed: keyserver error
The command "sudo -E apt-add-repository -y "ppa:ubuntu-toolchain-r/test"" failed and exited with 1 during .

**Comment by tklengyel** [comment]
@drakvuf-jenkins This is OK to test

File: src/main.cpp [hunk 1]
```diff
@@ -211,11 +214,17 @@ int main(int argc, char** argv)
...
+        INJECTION_TIMEOUT_OPTION = CHAR_MAX + 1,
+    };
+
     int long_index = 0;
     const option long_opts[] =
     {
-        {"rekall-kernel", optional_argument, NULL, 'r'},
-        {"rekall-tcpip", optional_argument, NULL, 'T'},
+        {"rekall-kernel", required_argument, NULL, 'r'},
+        {"rekall-tcpip", required_argument, NULL, 'T'},
```
[the remark this answers is not in the document]
Comment (disaykin) [reply src/main.cpp]: Because this options require argument.

[the remark this answers is not in the document]
Comment (tklengyel) [reply src/main.cpp]: Ah, ok, my bad - thought it means something else.

File: src/main.cpp [hunk 2]
```diff
@@ -211,11 +214,17 @@ int main(int argc, char** argv)
...
+        {"injection-timeout", required_argument, NULL, INJECTION_TIMEOUT_OPTION},
```
[the remark this answers is not in the document]
Comment (tklengyel) [reply src/main.cpp]: Go with `j`

[the remark this answers is not in the document]
Comment (disaykin) [reply src/main.cpp]: Done

**Comment by tklengyel** [comment]
@drakvuf-jenkins Retest this please

**Comment by tklengyel** [comment]
@drakvuf-jenkins Retest this please

**Comment by disaykin** [comment]
It looks like this error is not related to current changes. But I found and fix the use case that I missed earlier. In case of injector_start_app detects injected process before ntresumethread returns, then injector application will may leave stack in incorrect state.

**Comment by tklengyel** [comment]
@drakvuf-jenkins Retest this please

**Pull request merged**